test/unit/hooks/index.test.js

```diff
@@ -37,14 +37,123 @@ describe('Hooks', () => {
         () => plugin2
       )
 
-      const options = {}
-      const result = await Hooks.init(options)
+      const hookOptions = {
+        client: {
+          config () {}
+        },
+        context: null,
+        plugins: {}
+      }
+      const result = await Hooks.init(hookOptions)
 
       assert.typeOf(result.request, 'function')
       assert.typeOf(result.init, 'function')
       assert.typeOf(result.response, 'function')
     })
 
+    describe('per-request plugin options', function () {
+      beforeEach(function () {
+        suite.plugin1Stub = suite.sandbox.stub().returns({
+          request () {},
+          init () {}
+        })
+
+        suite.plugin2Stub = suite.sandbox.stub().returns({
+          request () {},
+          init () {}
+        })
+
+        GlobalConfig.plugins.push(
+          suite.plugin1Stub,
+          suite.plugin2Stub
+        )
+      })
+
+      it('should set per-request plugin options to false', async function () {
+        const hookOptions = {
+          client: {
+            config: () => ({
+              auth: {
+                clientId: '123',
+                clientSecret: 'abc'
+              }
+            })
+          },
+          context: null,
+          plugins: {
+            auth: false
+          }
+        }
+
+        await Hooks.init(hookOptions)
+
+        Sinon.assert.calledWith(suite.plugin1Stub, {
+          client: Sinon.match.any,
+          context: Sinon.match.any,
+          plugins: Sinon.match({
+            auth: false
+          })
+        })
+      })
+
+      it('should set per-request plugin options to false', async function () {
```

Comment:
This seems to be a duplicate of the one above?

Reply:
Thanks, I corrected the descriptions.